src/plugins/shape-generator/index.ts

```diff
@@ -321,9 +314,12 @@ export default class ShapeGenerator extends PluginController<{
     }
   }
 
-  onDispatch(e: DispatchedEvent) {
+  handleDispatchedAction(e: DispatchedEvent) {
     if (e.type === "toggle-add-expression") {
-      this.addExpressionMenuHandler();
+      // Wait for addExpressionBtn.ariaExpanded to be updated
+      queueMicrotask(() => {
```

Comment:
I haven't heard of `queueMicrotask` before. Neat!

It's not clear to me exactly when that JS callback is ran. Presumably after the dispatched action. But if one function call dispatched two actions in a single task, I don't know if it would run after both actions. There's the same problem with the `setTimeout(() => {}, 0)` speckled throughout the code. A more framework-brained way to do this is replacing `queueMicrotask(() => {})`` with `this.cc.runAfterDispatch(() => {})`, but I don't think you need to make this change.

Reply:
Ah, didn't know `runAfterDispatch` was a thing. I'll try it out and see if it works, it seems like exactly what I wanted and it also makes the code more readable.

Reply:
Wait, is it `this.cc.runAfterDispatch`? It doesn't seem to be typed...

Reply:
Ok in DevTools it seems to be `Calc.controller.runAfterDispatch`, and I tried that in my plugin and it seems to work. But yeah, the types are missing.